util/build_configs/setenv-example-3.bash

```diff
@@ -0,0 +1,247 @@
+##
+##  log_info "Start build_configs $dry_run $verbose $tools_targets"
+##
+##  $cwd/build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.tools.log" \
+##      --target-compiler=tools $tools_targets
+
+
+    # make clean
+
+    clean_targets="cleanall"
+
+    log_info "Start build_configs $dry_run $verbose $clean_targets"
+
+    $cwd/build_configs.py $dry_run $verbose -s $cwd/$setenv -l "$project.clean.log" \
+        --target-compiler=compiler $clean_targets
+
+    log_info "End $setenv"
+else
+
+# ===============
+# setenv callback
+# ===============
+
+    setenv=$( basename "${BASH_SOURCE[0]}" )
+    cwd=$( cd $(dirname "${BASH_SOURCE[0]}") && pwd )
+    source $cwd/functions.bash
+
+    log_info "Begin callback $setenv"
+    log_debug "with config=$BUILD_CONFIGS_CALLBACK"
+
+# =========================
+# setenv callback functions
+# =========================
+
+    source $cwd/module-functions.bash
+
+    # check module system
+    ck_module_list
+
+    if [ -n "$verbose" ] ; then
+        log_debug "Module list (sorted), after build_configs, before setenv:"
+        module list -t 2>&1 | tail -n +2 | sort
+    fi
+
+    gen_version_gcc=6.1.0
+    gen_version_intel=18.0.3.222
+    target_cpu_module=craype-sandybridge
+
+    function load_prgenv_gnu() {
+
+        local target_prgenv="PrgEnv-gnu"
+        local target_compiler="gcc"
+        local target_version=$gen_version_gcc
+
+        # unload any existing PrgEnv
+        unload_module_re PrgEnv-
+
+        # load target PrgEnv with compiler version
+        load_module $target_prgenv
+        load_module_version $target_compiler $target_version
+    }
+
+    function load_prgenv_intel() {
+
+        local target_prgenv="PrgEnv-intel"
+        local target_compiler="intel"
+        local target_version=$gen_version_intel
+
+        # unload any existing PrgEnv
+        unload_module_re PrgEnv-
+
+        # load target PrgEnv with compiler version
+        load_module $target_prgenv
+        load_module_version $target_compiler $target_version
+    }
+
+    function load_target_cpu() {
+
+        # legacy
+        unload_module perftools-base acml totalview atp cray-libsci xt-libsci
+        load_module cray-libsci
+
+        case "$1" in ( "" ) log_error "load_target_cpu missing arg 1"; exit 2;; esac
+        local target=$1
+
+        # target CPU: unload existing target-cpu modules, if any, then load our target-cpu
+        unload_module_re -v -network- craype-
+        load_module $target
+    }
+
+    export CHPL_MAKE_MAX_CPU_COUNT=24
+
+    # ---
+
+    # NOTE: The CHPL_TARGET_COMPILER env values from build_configs are not passed to Chapel.
+    # Instead, they drive module load commands in the setenv callback
+
+    case "$CHPL_TARGET_COMPILER" in
+    ( compiler)
+        load_prgenv_gnu
+        ;;
+    ( tools )
+        load_prgenv_gnu
+
+    ##  # If the installed libssl is too old to support TLS 1.1, some workarounds exist to
```

Comment:
Was this comment block left in by mistake or is this left as an example for how users can deal with outdated `libssl` issues?

Reply:
Yes, there were a couple places like this where I deliberately left some commented-out commands for that reason. Trying to have it both ways I guess- I wanted it to work the first time they tried it- which meant, basically, skip the python venv- yet provide some kind of guide towards getting the python venv to work.